packages/dev/core/src/FlowGraph/Blocks/Event/flowGraphReceiveCustomEventBlock.ts

```diff
@@ -22,19 +29,23 @@ export interface IFlowGraphReceiveCustomEventBlockConfiguration extends IFlowGra
 export class FlowGraphReceiveCustomEventBlock extends FlowGraphEventBlock {
     private _eventObserver: Nullable<Observer<any>>;
 
-    /**
-     * Output connection: The data sent with the event.
-     */
-    public eventData: FlowGraphDataConnection<any>;
-
     constructor(public config: IFlowGraphReceiveCustomEventBlockConfiguration) {
         super(config);
-        this.eventData = this.registerDataOutput("eventData", RichTypeAny);
+    }
+
+    public configure(): void {
+        super.configure();
+        for (let i = 0; i < this.config.eventData.length; i++) {
+            const dataName = this.config.eventData[i];
+            this.registerDataOutput(dataName, RichTypeAny);
+        }
     }
     public _preparePendingTasks(context: FlowGraphContext): void {
         const observable = context.configuration.coordinator.getCustomEventObservable(this.config.eventId);
-        this._eventObserver = observable.add((eventData) => {
-            this.eventData.setValue(eventData, context);
+        this._eventObserver = observable.add((eventDatas: any[]) => {

```

Comment:
is this being removed on dispose?

Reply:
Yep, it's removed on _cancelPendingTasks, which is called when we dispose of a graph and its blocks